astroid/exceptions.py

```diff
@@ -290,6 +290,22 @@ def __init__(self, target: "nodes.NodeNG") -> None:
         )
 
 
+class FrameMissing(ParentMissingError):
+    """Raised when a call to node.frame() does not return a node. This is because
+    a node in the chain does not have a parent attribute and therefore does not
+    return a node for frame().
+
+    Standard attributes:
+        target: The node for which the parent lookup failed.
+    """
+
+    def __init__(self, target: "nodes.NodeNG") -> None:
+        # pylint: disable-next=bad-super-call
+        super(ParentMissingError, self).__init__(
+            message=f"Frame not found on {target!r}"
+        )
```

Comment:
I'm wondering if it's actually necessary to create a new exception subclass. All existing `frame()` and `scope()` methods raise just a `ParentMissingError`.

@DanielNoord I don't actually remember the exact details why we decided to add `StatementMissingError` maybe you do? Was there an actual need to have a more specific error or was it just an idea that we adopted?

Reply:
I think this followed from the logic that we were catching the `AttributeError` ourselves so it sort of naturally followed that we had to create a new error to catch. 
It might be a bit overkill, but I think it makes sense to make a new subclass: the more fine grained the error is the better users will know how to handle it.